src/SimpleMemSystem.cpp

```diff
@@ -153,16 +154,25 @@ void SimpleMemSystem::load_segments() {
 
                 mem_msg << "The memory segment " <<  i  <<  " is initialized with 0x" << std::hex << length << " bytes from input_image !";
                 etiss::log(etiss::INFO, mem_msg.str());
-            } else if (initString != "") {
-                mem_msg << "The memory segment " <<  i  <<  " is initialized with 0x" << std::hex << length << " elements with value: " << initString;
-                etiss::log(etiss::INFO, mem_msg.str());
+            } else if (initEleSet) {
+                if (initString.find("random") == 0 || initString.find("RANDOM")== 0) {
+                    mem_msg << "The memory segment " <<  i  <<  " is initialized with 0x" << std::hex << length << " random bytes !";
+                    etiss::log(etiss::INFO, mem_msg.str());
+                } else if (initString.find("0x") == 0) {
+                    mem_msg << "The memory segment " <<  i  <<  " is initialized with 0x" << std::hex << length << " elements with hex value: " << initString;
+                    etiss::log(etiss::INFO, mem_msg.str());
+                }
+                else
+                {
+                    mem_msg << "The memory segment " <<  i  <<  " is initialized with 0x" << std::hex << length << " elements with the string: " << initString;
+                    etiss::log(etiss::INFO, mem_msg.str());
+                }
             } else {
-                mem_msg << "The memory segment " <<  i  <<  " is initialized with 0x" << std::hex << length << " random values !";
+                mem_msg << "The memory segment " <<  i  <<  " is allocated uninitialized with length 0x" << std::hex << length << " !";
                 etiss::log(etiss::INFO, mem_msg.str());
             }
```

Comment:
all these conditionals do is print debug messages, move the messages to `MemSegment()` where these differentiations have to be made anyways

Reply:
agreed, `i` should not be necessary to keep here.